source/PlayerInfo.cpp

```diff
@@ -2354,7 +2356,35 @@ void PlayerInfo::StepMissions(UI *ui)
 		else if(mission.CanComplete(*this))
 			RemoveMission(Mission::COMPLETE, mission, ui);
 		else if(mission.Destination() == GetPlanet() && !freshlyLoaded)
+		{
 			mission.Do(Mission::VISIT, *this, ui);
+			
+			// On visit dialogs are handled separately as to avoid a player
+			// getting spammed by on visit dialogs if they are stacking jobs
+			// from the same destination.
+			map<Mission::Trigger, MissionAction> actions = mission.GetActions();
```

Comment:
I'd use `auto` here.

Perhaps `GetActions` should be `GetAction` and take an argument for the action type to retrieve. It would need to return a pointer to the MissionAction, which you would null-check in this method.